pkg/apis/sources/v1alpha1/gcp_pubsub_validation.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/knative/pkg/apis"
+)
+
+func (current *GcpPubSubSource) CheckImmutableFields(og apis.Immutable) *apis.FieldError {
```

Comment:
I did not catch this the first time around, this will not work for eventing sources unless you make and install the webhook cmd from pkg. Eventing sources does not have the pkg webhook, this type of validation should be done int eh openapi contract in the CRD.

Reply:
Good catch @n3wscott, we'll have to run the webhook for this to work. Seems like an enhancement for a different PR. I'm fine with leaving this code in even if it's dead since adding the webhook seems like it could follow soon after. I'm also fine with removing it.

> this type of validation should be done int eh openapi contract in the CRD.

According to https://github.com/kubernetes/kubernetes/issues/65973 Kubernetes openapi doesn't support immutability yet.

Reply:
Created https://github.com/knative/eventing-sources/issues/59.